feature/experimental/policy/policy_vrf_selection/ate_tests/base_vrf_selection/README.md

```diff
@@ -2,24 +2,86 @@
-    *   Match IPinIP, DSCP 42, 46 - routed to network-instance 10, ensure that DSCP 42 and 46 packets are received at VLAN
-        Ensure that DSCP 0 packets are not received at VLAN 10.
+Test different VRF selection policies.
+
+## Topology
+
+*   Connect ATE Port-1 to DUT Port-1, and ATE Port-2 to DUT Port-2.
+
+*   Configure DUT with ingress interface as Port-1 and egress interface as Port-2. DUT Port-2 is connected to ATE Port-2 and is configured for VLAN-10 and VLAN-20 such that sub-interfaces Port-2.10 and Port-2.20 are part of VLAN 10 and VLAN 20 respectively.
+
+*   Configure network-instance “VRF-10” of type L3VRF and assign Port2.10 to network-instance "VRF-10" 
+        
+*   Configure network-instance “DEFAULT” of type DEFAULT_INSTANCE. DUT-Port1 and DUT Port-2.20 should be part of "DEFAULT" Network w/o requiring 
+    explicit configuration.
+
+## Procedure
+
+*   Test-Case 1
+
+        *   Configure DUT to match on IPinIP protocol (protocol number 4) in the outer IPv4 header and punt it to 
+            network-instance VRF-10. All other traffic should be punted to the Default VRF. These will be, native IPv4, 
+            native IPv6 and IPv6inIP (protocol 41 in the outer IPv4 header) traffic.
+        
+        *   Start all traffic flows defined in the section "Flows" below and validate packet forwarding.
+
+*   Test-Case 2
+
+        *   Configure DUT to match on IPinIP protocol (protocol number 4 in the outer IPv4 header) with specific outer 
+            IPv4 source address as "222.222.222.222" and punt it to network-instance VRF-10. All other traffic should be 
+            punted to the Default VRF. These will be, IPinIP w/o source as "222.222.222.222", native IPv4, native IPv6 and 
+            IPv6inIP (protocol 41 in the outer IPv4 header) traffic.
+
+        *   Start all traffic flows defined in the section "Flows" below and validate packet forwarding.   
+
+
+*   Test-Case 3
+
+        *   Configure DUT to match on IPv6inIP protocol (protocol number 41 in the outer IPv4 header) and punt it to 
+            network-instance VRF-10. All other traffic should be punted to the Default VRF. These will be, native IPv4, 
+            native IPv6 and IPinIP (protocol 4 in the outer IPv4 header) traffic.
+
+        *   Start all traffic flows defined in the section "Flows" below and validate packet forwarding.        
+
+* Test-Case 4
+          
+        *  Configure DUT to match on IPv6inIP protocol (protocol number 41 in the outer IPv4 header) with specific 
+           outer IPv4 source address "222.222.222.222" and punt it to the network-instance VRF-10. 
+           All other traffic should be punted to the Default VRF. These will be, IPv6inIP w/o source as 
+           "222.222.222.222", native IPv4, native IPv6 and IPinIP (protocol 4 in the outer IPv4 header) traffic.
+
+        *   Start all traffic flows defined in the section "Flows" below and validate packet forwarding. 
+
+## Flows
```

Comment:
Now I see the downside of using interface IP as the destination IPs. Because the same flow can go to different port depends on the VRF selection policy configs, basically it requires the code to adjust the flow setup in each test. This probably will make the code hard to read. 

I think it's better to stick to a single (static) destination IP behind the ATE port, so that you only need to setup once. 

WDYT?

Reply:
Updated. Makes sense!

Reply:
As discussed, we will have to use the original model of selecting specific flows for different tests to ensure the policies are tested accurately otherwise the tests will always pass masking vendor bugs if any.